pkg/client/observatorium/client_test.go

```diff
@@ -0,0 +1,358 @@
+package observatorium
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	pAPI "github.com/prometheus/client_golang/api"
+	pV1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+var (
+	observabilityConfiguration = NewObservabilityConfigurationConfig()
+	configuration              = Configuration{
+		Cookie:   observabilityConfiguration.Cookie,
+		Timeout:  observabilityConfiguration.Timeout,
+		Debug:    observabilityConfiguration.Debug,
+		Insecure: observabilityConfiguration.Insecure,
+		AuthType: "test",
+		BaseURL:  "",
+	}
+	testValue = "test"
+)
+
+func Test_NewObservatoriumClient(t *testing.T) {
+	type args struct {
+		c *ObservabilityConfiguration
+	}
+
+	tests := []struct {
+		name     string
+		modifyFn func(config *ObservabilityConfiguration)
+		args     args
+		wantErr  bool
+	}{
+		{
+			name: "should not return an error when providing default ObservabilityConfiguration",
+			args: args{
+				c: NewObservabilityConfigurationConfig(),
+			},
+			wantErr: false,
+		},
+		{
+			name: "should not return an error when providing default ObservabilityConfiguration with AuthType = AuthTypeSso",
+			args: args{
+				c: NewObservabilityConfigurationConfig(),
+			},
+			modifyFn: func(config *ObservabilityConfiguration) {
+				config.AuthType = AuthTypeSso
+			},
+			wantErr: false,
+		},
+		{
+			name: "should not return an error when providing default ObservabilityConfiguration with mock enabled",
+			args: args{
+				c: NewObservabilityConfigurationConfig(),
+			},
+			modifyFn: func(config *ObservabilityConfiguration) {
+				config.EnableMock = true
+			},
+			wantErr: false,
+		},
+	}
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := tt.args.c
+			if tt.modifyFn != nil {
+				tt.modifyFn(config)
+			}
+			_, err := NewObservatoriumClient(config)
+			Expect(err != nil).To(Equal(tt.wantErr))
```

Comment:
there is no test case where `tt.wantErr` is true

Reply:
It is already returning an error.
I believe the confusion comes from the fact that it is using `named return values`.
With the `return` in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/observatorium/client.go#L65 it is returning whatever values are stored in `client` and `err`.